src/libs/actions/Report.js

```diff
@@ -1225,6 +1157,107 @@ Onyx.connect({
     callback: handleReportChanged,
 });
 
+/**
+ * Deletes a comment from the report, basically sets it as empty string
+ *
+ * @param {Number} reportID
+ * @param {Object} reportAction
+ */
+function deleteReportComment(reportID, reportAction) {
+    const sequenceNumber = reportAction.sequenceNumber;
+
+    const optimisticReportActions = {
+        [sequenceNumber]: {
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+        },
+    };
+    const optimisticReport = {
+        lastMessageText: ReportActions.getLastVisibleMessageText(reportID, {
+            [sequenceNumber]: {
+                ...reportAction,
+            },
+        }),
+    };
+
+    // If the deleted comment is more recent than our last read comment, update the unread count
+    if (sequenceNumber > getLastReadSequenceNumber(reportID)) {
+        optimisticReport.unreadActionCount = Math.max(getUnreadActionCount(reportID) - 1, 0);
+    }
+
+    // List of optmistic data to change
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            value: optimisticReportActions,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+            value: optimisticReport,
+        },
+    ];
+
+    // On Success remove the message content and pendingAction state
+    const successData = {
+        [sequenceNumber]: {
+            pendingAction: null,
+            message: [{
+                html: '',
+                text: '',
+            }],
```

Comment:
Removing the pendingAction is good.
But I think the message part is not necessary and maybe even bad. 

The server should be returning this already via Pusher so this is redundant

https://github.com/Expensify/Web-Expensify/blob/9de61ce83ed42e60f0bede48577d855b4fea2119/lib/ReportAPI.php#L483-L495